src/libexpr/eval-settings.hh

```diff
@@ -19,11 +19,26 @@ struct EvalSettings : Config
     Setting<Strings> nixPath{
         this, getDefaultNixPath(), "nix-path",
         R"(
-          List of directories to be searched for `<...>` file references
-
-          In particular, outside of [pure evaluation mode](#conf-pure-eval), this determines the value of
+          List of search paths to use for lookup paths.
+          This setting determines the value of
           [`builtins.nixPath`](@docroot@/language/builtin-constants.md#builtins-nixPath).
-        )"};
+
+          The default value is
+
+          ```
+          $HOME/.nix-defexpr/channels
+          nixpkgs=$NIX_STATE_DIR/profiles/per-user/root/channels/nixpkgs
+          $NIX_STATE_DIR/profiles/per-user/root/channels
+          ```
+
+          It can be overridden with the [`NIX_PATH` environment variable](@docroot@/command-ref/env-common.md#env-NIX_PATH) or the [`-I` command line option](@docroot@/command-ref/opt-common.md#opt-I).
+
+          > **Note**
+          >
+          > If [restricted evaluation](#conf-restrict-eval) is enabled, the default value is ignored.
```

Comment:
What do you mean? I get
```shell-session
$ NIX_PATH=foo=bar nix-instantiate --restrict-eval --expr 'builtins.nixPath' --eval
warning: Nix search path entry 'bar' does not exist, ignoring
[ { path = "bar"; prefix = "foo"; } ]
```

Reply:
I mean the default hard-coded value

Reply:
I am still confused?

```shell-session
$ unset NIX_PATH
$ NIX_CONF_DIR='/foo' nix-instantiate --store 'dummy://' --restrict-eval --expr 'builtins.nixPath' --eval
[ { path = "/home/jcericson/.nix-defexpr/channels"; prefix = ""; } { path = "/nix/var/nix/profiles/per-user/root/channels/nixpkgs"; prefix = "nixpkgs"; } { path = "/nix/var/nix/profiles/per-user/root/channels"; prefix = ""; } ]
```

Reply:
Pretty sure this is an instance of https://github.com/NixOS/nix/issues/9574, because `restrict-eval` ignoring `nix-path` default values was supposed to be the intended behaivor https://github.com/NixOS/nix/pull/7871#issuecomment-1446418480. I removed this for now, so we don't document things which are not the case.